tsdb/block.go

```diff
@@ -476,15 +477,15 @@ func (r blockIndexReader) LabelValues(name string, matchers ...*labels.Matcher)
 		return st, errors.Wrapf(err, "block: %s", r.b.Meta().ULID)
 	}
 
-	return labelValuesWithMatchers(r.ir, name, matchers...)
+	return labelValuesWithMatchers(context.Background(), r.ir, name, matchers...)
```

Comment:
Does it make sense to add ctx to the function also and pass here?

Reply:
Yeah it does. But i did not do it cause i did not want to introduce a breaking change on the [IndexReader](https://github.com/prometheus/prometheus/blob/5c5fa5c319fca713506fa144ec6768fddf00d466/tsdb/block.go#L60) interface. 

Im happy to do it if we think is OK to change the `IndexReader` interface and add the ctx as parameter to all the functions (what i think make lots sense).

Reply:
If not done yet, but it makes sense, then these should be `context.TODO()` instead of `context.Background()`.

Reply:
I have a [PR](https://github.com/prometheus/prometheus/pull/12665) for adding a context argument to LabelValues and SortedLabelValues. Agree with @colega this should be `context.TODO()`.